broadinstitute/scp-ingest-pipeline
# Enable pairwise DE in ingest pipeline (SCP-5766)

#### BACKGROUND & CHANGES

This update adds pairwise DE to ingest pipeline. A new `--differential-expression` ingest pipeline parameter, `--de-type` is instantiated as "rest" by default, allowing existing "one vs rest" invocations to remain unchanged. Pairwise invocations add two additional parameters `--group1` and `--group2`. ~~Scanpy's rank_genes_groups function will accept a list of annotation values for "--group1", while we won't be taking advantage of that capability, it does require `--group1` to be passed as a list~~. Both `--group1` and `--group2` ~~is~~ passed as ~~a simple~~ string.

#### MANUAL TESTING

Testing using ingest_pipeline CLI requires an AnnData file and cluster & metadata AnnData fragments extracted from the AnnData file. 
1. Download the following files (you may already have them from PR#372):
	**HTAPP-330-SMP-1082_compliant.h5ad**: gs://fc-2f8ef4c0-b7eb-44b1-96fe-a07f0ea9a982/test_Data/differential_expression/HTAPP-330-SMP-1082/HTAPP-330-SMP-1082_compliant.h5ad
	**HTAPP-330-SMP-1082_h5ad_frag.cluster.X_umap.tsv.gz**: gs://fc-2f8ef4c0-b7eb-44b1-96fe-a07f0ea9a982/test_Data/differential_expression/HTAPP-330-SMP-1082/HTAPP-330-SMP-1082_h5ad_frag.cluster.X_umap.tsv.gz (Note: file will lose the .gz suffix BUT will still need to be decompressed :(
	**HTAPP-330-SMP-1082_h5ad_frag.metadata.tsv.gz**: gs://fc-2f8ef4c0-b7eb-44b1-96fe-a07f0ea9a982/test_Data/differential_expression/HTAPP-330-SMP-1082/HTAPP-330-SMP-1082_h5ad_frag.metadata.tsv.gz (Note: file will lose the .gz suffix BUT will still need to be decompressed :(1. Initialize your dev environment and ```export BYPASS_MONGO_WRITES='yes'```
1. Run the following ingest pipeline job: 
`python ingest_pipeline.py --study-id addedfeed000000000000000 --study-file-id dec0dedfeed1111111111111 differential_expression --annotation-name leiden --de-type pairwise --group1 "1" --group2 "2" --annotation-type group --annotation-scope study --annotation-file HTAPP-330-SMP-1082_h5ad_frag.metadata.tsv.gz --cluster-file HTAPP-330-SMP-1082_h5ad_frag.cluster.X_umap.tsv.gz --cluster-name umap --matrix-file-path HTAPP-330-SMP-1082_compliant.h5ad  --matrix-file-type h5ad --study-accession SCPdev --differential-expression`
1. Confirm that the job generates "umap--leiden--1--2--study--wilcoxon.tsv"
1. [optional] Compare your output to the reference output file in the reference google bucket at gs://fc-2f8ef4c0-b7eb-44b1-96fe-a07f0ea9a982/test_Data/differential_expression/HTAPP-330-SMP-1082/umap--leiden--1--2--study--wilcoxon.tsv
